explainaboard/processors/processor_registry.py

```diff
@@ -3,9 +3,89 @@
 from __future__ import annotations
 
 from explainaboard import TaskType
+from explainaboard.processors.argument_pair_extraction import (
+    ArgumentPairExtractionProcessor,
+)
+from explainaboard.processors.aspect_based_sentiment_classification import (
+    AspectBasedSentimentClassificationProcessor,
+)
+from explainaboard.processors.chunking import ChunkingProcessor
+from explainaboard.processors.cloze_generative import ClozeGenerativeProcessor
+from explainaboard.processors.cloze_multiple_choice import ClozeMultipleChoiceProcessor
+from explainaboard.processors.conditional_generation import (
+    ConditionalGenerationProcessor,
+)
+from explainaboard.processors.extractive_qa import QAExtractiveProcessor
+from explainaboard.processors.grammatical_error_correction import (
+    GrammaticalErrorCorrectionProcessor,
+)
+from explainaboard.processors.kg_link_tail_prediction import (
+    KGLinkTailPredictionProcessor,
+)
+from explainaboard.processors.language_modeling import LanguageModelingProcessor
+from explainaboard.processors.machine_translation import MachineTranslationProcessor
+from explainaboard.processors.named_entity_recognition import NERProcessor
+from explainaboard.processors.nlg_meta_evaluation import NLGMetaEvaluationProcessor
 from explainaboard.processors.processor import Processor
+from explainaboard.processors.qa_multiple_choice import QAMultipleChoiceProcessor
+from explainaboard.processors.qa_open_domain import QAOpenDomainProcessor
+from explainaboard.processors.qa_tat import QATatProcessor
+from explainaboard.processors.summarization import SummarizationProcessor
+from explainaboard.processors.tabular_classification import (
+    TabularClassificationProcessor,
+)
+from explainaboard.processors.tabular_regression import TabularRegressionProcessor
+from explainaboard.processors.text_classification import TextClassificationProcessor
+from explainaboard.processors.text_pair_classification import (
+    TextPairClassificationProcessor,
+)
+from explainaboard.processors.word_segmentation import CWSProcessor
 
-_processor_registry: dict = {}
+
+def get_processor_class(task: TaskType) -> type[Processor]:
+    """Returns a Processor class from the given task type.
+
+    Args:
+        task: A task type.
+
+    Returns:
+        A Processor class associated with the given task type.
+
+    Raises:
+        ValueError: if the given task is not supported.
+        TypeError: if the obtained class is not a subclass of Processor class.
+    """
+    task_to_procesor = {
```

Comment:
This dict is created every time this function is called. Place it onto the top-level scope.

Reply:
Done